### PR TITLE
RHMAP-7436 Added more API tests

### PR DESCRIPTION
**Pull request by psturc**
## Motivation

https://issues.jboss.org/browse/RHMAP-7436
## Checklist
- [x] Secure API
- [x] Hash API
- [x] Host API
## Ping

@AdamSaleh @matskiv @thradec

### CODE REVIEW

File: lib/hash.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+var fh = require('fh-mbaas-api');
+var express = require('express');
+var util = require('./util');
+
+function getOptionsForAlgorithm(algorithm) {
+  return { 'algorithm': algorithm, 'text': 'Text to be hashed.' };
```
Comment (thradec) [inline lib/hash.js]: I would take the value from request parameter and use this only as default

File: lib/hash_tests.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+function getOptionsForAlgorithm(algorithm) {
+  return { 'algorithm': algorithm, 'text': 'Text to be hashed.' };
+}
+
+describe('Hash API', function() {
+  it('should generate a hash value for given input with MD5 algorithm', function(done) {
+    fh.hash( getOptionsForAlgorithm('MD5'), function(err, res) {
+      expect(err).to.not.exist;
+      expect(res.hashvalue).to.be.a('string');
```
Comment (thradec) [inline lib/hash_tests.js]: how long is the hashed value, can we add more specific assertions?

File: lib/secure.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+var fh = require('fh-mbaas-api');
+var express = require('express');
+var winston = require('winston');
```
Comment (thradec) [inline lib/secure.js]: is not used

**Comment by thradec** [comment]
LGTM
added only minor comments
plus add those new endpoints into readme, thanks

**Comment by psturc** [comment]
@thradec Thanks for the review! More tests should be added later.

File: lib/hash_tests.js (hunk 2 above)
Comment (psturc) [inline lib/hash_tests.js]: Sure, good point.

File: lib/hash_tests.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+function performHashCheck(algorithm, length, done) {
+  fh.hash({
+    'algorithm': algorithm,
+    'text': 'Text to be hashed.'
+  }, function(err, res) {
+    try {
+      expect(err).to.not.exist;
+      expect(res.hashvalue)
+        .to.be.a('string')
+        .to.have.lengthOf(length);
```
Comment (david-martin) [inline lib/hash_tests.js:19]: Should we do an equals check to verify the hashed value is as expected? e.g. use a fixed string and compare the output of fh.hash to what we expect the hashed value to be?

**Comment by david-martin** [comment]
These tests look good @psturc :+1: 
Only 1 comment above about the fh.hash test

File: lib/hash_tests.js (hunk 4 above)
Comment (psturc) [inline lib/hash_tests.js:19]: Yes, this is a good point. Will provide a link to PR with this test enhancement.

Update: https://github.com/feedhenry/testing-cloud-app/pull/7

**Pull request merged**